syntax = "proto3";


// The greeter service definition.
service Greeter {
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply) {}
  }
  
  // The request message containing the user's name.
  message HelloRequest {
    string name = 1;
  }
  
  // The response message containing the greetings
  message HelloReply {
    string message = 1;
  }

// The training service definition.
service ModelTrainingService {
  // Sends training settings and dataset to Python for model training
  rpc TrainModel(TrainRequest) returns (TrainResponse);
}

// The request message containing the training settings and dataset.
message TrainRequest {
  string name = 1;
  string tag = 2;
  string algorithm = 3;
  string blobName = 4;
  int32 minCount = 5;
  int32 vectorSize = 6;
  int32 windowSize = 7;
  int32 epochs = 8;
}

// The response message containing the training results.
message TrainResponse {
  string message = 1;
  string taskId = 2;
}

// python3 -m grpc_tools.protoc -I./protobuf --python_out=./generated --grpc_python_out=./generated ./protobuf/embeddings.proto
